### ISSUE TITLE
output of lfs.dir in luatex

### ISSUE THREAD

**Issue by u-fischer** [MiKTeX/miktex#762]
- [x] I have read and acknowledged this HOWTO article:  
  [Submitting a MiKTeX problem report](https://miktex.org/howto/bug-report)

In miktex on windows 10  `lfs.dir` doesn't report the two standard directories 

~~~~
\documentclass{article}
\begin{document}
\directlua{ texio.write("TESTDIR ") for entry in lfs.dir("somedir") do texio.write("entry: " .. entry) end}
abc
\end{document}
~~~~

gives (with somedir an existing, empy direktory )

~~~~
TESTDIR 
~~~~

In texlive I get 

~~~~
TESTDIR entry: .entry: ..
~~~~

This breaks the expections of l3build which therefor currently works in miktex. 

Is this difference intended? If yes l3build would have to work around this.

**Comment by edocevoli** [comment]
MiKTeX doesn't return '.' and '..'. This is a bug.

But I am not sure that your test is 100% reliable, because not all OS (e.g., OpenVMS) have the notion of '.' and '..'.

**Comment by edocevoli** [comment]
Thank you, this has been fixed with v21.6.
